libraries/react-experimental/src/basic-tests.js

```diff
@@ -0,0 +1,179 @@
+      expect(wc).to.exist;
+      let shadowRoot = wc.shadowRoot;
+      let heading = shadowRoot.querySelector("h1");
+      expect(heading).to.exist;
+      expect(heading.textContent).to.eql("Test h1");
+      let paragraph = shadowRoot.querySelector("p");
+      expect(paragraph).to.exist;
+      expect(paragraph.textContent).to.eql("Test p");
+    }
+
+    it("can display a Custom Element with children in a Shadow Root", function() {
+      this.weight = 3;
+      let root = ReactDOM.render(<ComponentWithChildren />, scratch);
+      let wc = root.wc;
+      expectHasChildren(wc);
+    });
+
+    it("can display a Custom Element with children in a Shadow Root and pass in Light DOM children", async function() {
+      this.weight = 3;
+      let root = ReactDOM.render(<ComponentWithChildrenRerender />, scratch);
+      let wc = root.wc;
+      await Promise.resolve();
+      expectHasChildren(wc);
+      expect(wc.textContent.includes("2")).to.be.true;
+    });
+
+    it("can display a Custom Element with children in the Shadow DOM and handle hiding and showing the element", function() {
+      this.weight = 3;
+      let root = ReactDOM.render(<ComponentWithDifferentViews />, scratch);
+      let wc = root.wc;
+      expectHasChildren(wc);
+      root.toggle();
+      let dummy = ReactDOM.findDOMNode(root.refs.dummy);
+      expect(dummy).to.exist;
+      expect(dummy.textContent).to.eql("Dummy view");
+      root.toggle();
+      wc = root.wc;
+      expectHasChildren(wc);
+    });
+  });
+
+  describe("attributes and properties", function() {
+    it("will pass boolean data as either an attribute or a property", function() {
+      this.weight = 3;
+      let root = ReactDOM.render(<ComponentWithProperties />, scratch);
+      let wc = root.wc;
+      let data = wc.bool || wc.hasAttribute("bool");
+      expect(data).to.be.true;
+    });
+
+    it("will pass numeric data as either an attribute or a property", function() {
+      this.weight = 3;
+      let root = ReactDOM.render(<ComponentWithProperties />, scratch);
+      let wc = root.wc;
+      let data = wc.num || wc.getAttribute("num");
+      expect(parseInt(data, 10)).to.eql(42);
+    });
+
+    it("will pass string data as either an attribute or a property", function() {
+      this.weight = 3;
+      let root = ReactDOM.render(<ComponentWithProperties />, scratch);
+      let wc = root.wc;
+      let data = wc.str || wc.getAttribute("str");
+      expect(data).to.eql("React");
+    });
+
+    // TODO: Is it the framework's responsibility to check if the underlying
+    // property is defined? Or should it just always assume it is and do its
+    // usual default behavior? Preact will actually check if it's defined and
+    // use an attribute if it is not, otherwise it prefers properties for
+    // everything. Is there a "right" answer in this situation?
+
+    // it('will set boolean attributes on a Custom Element that has not already been defined and upgraded', function() {
```

Comment:
What about using `xit` rather than commenting these out?

Reply:
Oh wait, is this still `mocha` under the hood? You might have to use `this.skip()` in the test instead.

Reply:
These tests are commented out in all the repos, will delete in a follow up